index.d.ts

```diff
@@ -0,0 +1,30 @@
+export interface Options {
+	/**
+	 * String to use as replacement for reserved filename characters.
+	 *
+	 * Cannot contain: `<` `>` `:` `"` `/` `\` `|` `?` `*`
+	 *
+	 * @default '!'
+	 */
+	readonly replacement?: string;
+}
+
+/**
+ * Accepts a filename and returns a valid filename.
+ *
+ * @param input - A string to convert to a valid filename.
+ */
+export interface Filenamify {
+	(input: string, options?: Options): string;
+
+	/**
+	 * Accepts a path and returns the path with a valid filename.
+	 *
+	 * @param input - A string to convert to a valid path with a filename.
+	 */
+	path(input: string, options?: Options): string;
+}
+
+declare const filenamify: Filenamify;
```

Comment:
I'm just curious. Why define an interface here instead of just inlining it? I can't really see the benefit of exporting a named interface.

```ts
declare const filenamify: {
	(input: string, options?: Options): string;

	/**
	 * Accepts a path and returns the path with a valid filename.
	 *
	 * @param input - A string to convert to a valid path with a filename.
	 */
	path(input: string, options?: Options): string;
};
```

Reply:
An oversight, not really needed.